### ISSUE TITLE
Feature request: quick filter auto close

### ISSUE THREAD

**Issue by Eralmidia** [IgniteUI/igniteui-angular#2979]
### Description  
Just wanted to say good job on the new quick filter implementation. I have one tiny request, however. Could we have an option, so that the filter would auto close if it looses focus? If you use the filter a lot, it starts to feel a bit cumbersome to always have to move the mouse all the way over to the right side of the screen (where the close button is). 

It's kind of the same reason many popups/dialog solutions lets you click anywhere outside the dialog, instead of having to target the close button.

 * igniteui-angular version: 6.2.0

**Comment by bkulov** [comment]
@Eralmidia We were thinking of closing the filtering edit mode on Esc key press. That way you still could interact with the grid (move, resize, group, sort columns, etc.) and have the filtering edit row opened. And pressing Esc key anywhere in the filtering edit row (except if there is a dropdown opened) would close it.

**Comment by StefanIvanov** [comment]
Good idea @bkulov since the Esc key will spare some pressing of the Tab key, however I believe @Eralmidia has a good point that if the row loses focus we should be considerate of hiding the filtering. The case may not be once the user tabs out of the filtering dialog but rather once a feature is triggered e.g. sorting a column or editing a cell in the body. How does that sound @Eralmidia ?

**Comment by bkulov** [comment]
@Eralmidia We've discussed the behavior of the filtering row. In our opinion it's good to have the filtering edit row opened even when doing other operations in the grid (column grouping, resizing, moving, sorting, cell editing, etc.). It's a common behavior in other grids to have quick filtering row enabled/active all the time. We think that even though our filtering UI is a bit different it's still convenient to quickly change the filtering conditions when you collapse/expand grouped rows, scroll or do other operations in the grid.

We've already implemented closing filtering edit mode with Esc key. If you don't have other remarks I'll close this issue.
